libs/newsletter/src/driver/hubspot/newsletter.service.spec.ts

```diff
@@ -0,0 +1,139 @@
+  });
+  it('should be created', () => {
+    expect(newsletterService).toBeTruthy();
+  });
+
+  describe('sending a newsletter submission to hubspot', () => {
+    it('should take Hubspot configuration from the module configuration', async(inject([HttpClient, DaffNewsletterConfig], (http, config) => {
+      spyOn(http, 'post').and.callThrough();
+      const submission = { email: 'email@email.com', firstName: 'John', lastName: 'Doe' };
+      const version = config.version === undefined ? 'v3' : config.version;
+      const submissionString = 'https://api.hsforms.com/submissions/' + version + '/integration/submit/'
+        + config.portalId + '/' + config.guid
+      newsletterService.send(submission).subscribe(() => {
+        expect(http.post).toHaveBeenCalledWith(submissionString, jasmine.any(Object));
+      });
+
+      const req = httpMock.expectOne('https://api.hsforms.com/submissions/v3/integration/submit/undefined/undefined');
+
+      req.flush('');
+    })));
+    it('should transform a request send to hubspot', async(inject([DaffNewsletterTransformer], (transformer) => {
+      spyOn(transformer, 'transformOut').and.callThrough();
+      const submission = { email: 'email@email.com', firstName: 'John', lastName: 'Doe' };
+      newsletterService.send(submission).subscribe(() => {
+        expect(transformer.transformOut).toHaveBeenCalled();
+      });
+      const req = httpMock.expectOne('https://api.hsforms.com/submissions/v3/integration/submit/undefined/undefined');
+
+      req.flush('');
+    })));
+    it('should transform the response recieved from Hubspot', async(inject([DaffNewsletterTransformer], (transformer) => {
+      spyOn(transformer, 'transformIn').and.callThrough();
+      const submission = { email: 'email@email.com', firstName: 'John', lastName: 'Doe' };
+      newsletterService.send(submission).subscribe(() => {
+        expect(transformer.transformIn).toHaveBeenCalled();
+      });
+      const req = httpMock.expectOne('https://api.hsforms.com/submissions/v3/integration/submit/undefined/undefined');
+
+      req.flush('');
+    })));
+    it('should include a the hubspot UTK in the request, if it hubspot cookie exists', () => {
+      doc.cookie = 'hubspotutk=mycookie;'
+      const submission = { email: 'email@email.com', firstName: 'John', lastName: 'Doe' };
+      newsletterService.send(submission).subscribe();
+
+      const req = httpMock.expectOne((testReq: HttpRequest<any>) => {
+        expect(testReq.url).toEqual('https://api.hsforms.com/submissions/v3/integration/submit/undefined/undefined');
+        expect(testReq.body.context.hutk).toEqual('mycookie');
+        return testReq.url === 'https://api.hsforms.com/submissions/v3/integration/submit/undefined/undefined' && testReq.body.context.hutk === 'mycookie';
+      });
+
+      req.flush('');
+    });
+    it('should not include a the hubspot UTK in the request, if the hubspot cookie exists', () => {
```

Comment:
`should not include the hubspot UTK in the request, if the hubspot cookie exists`

Reply:
Addressed in 7aeb11b1ed22f3bb2f03cf9a1023156c2824ca5b